app/src/main/java/com/simonramstedt/yoke/YokeActivity.java

```diff
@@ -709,23 +712,43 @@ public void onStopDiscoveryFailed(String serviceType, int errorCode) {
         mNsdManager.stopServiceDiscovery(this);
     }
 
-    private void closeConnection() {
+    // closeConnection(true) sends a copy of the current status with its first byte changed to 0xFF
+    // before closing the connection. This signals the PC client to disconnect the device.
+    // (Enforcing the same length on the message as the previous one seems to be required on Windows.)
+    //
+    // closeConnection(false) closes the connection with no warning: it's intended for both
+    // accidental disconecttions (like exiting the app) and error states (where it doesn't make
+    // sense to warn the client and doing so would probably send to a null socket or cause more errors.)
+    private void closeConnection(boolean warn) {
```

Comment:
I'm not sure whether it's better to keep this function this way, or to split it into 2 functions (`requestDisconnect` and `closeSocket`). Boolean flags are usually not that clear:

```java
this.requestDisconnect();
this.closeSocket();
```

vs

```java
this.closeConnection(true);
```

Plus, the comment makes it sound like these are 2 separate functions, though that's just my opinion.

Reply:
Your proposal is much more clear (and cleaner), I'll push a commit to adopt it.

I'll also push a commit to both PRs to move the disconnection pattern to the JavaScript side. It's much more useful and tunable than hardcoding it in the app.

Reply:
Commits 3c9459460ea07e9c21b9fc15112814ab28319e6a and 6b4cc04693b53747d4160156d593e1d3a0335d15

And commit 199e5d5b9469b91b523aa715e49e8f2b0c0ce9cc on the other repo.